managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java

```diff
@@ -134,6 +141,7 @@ public Result delete(UUID customerUUID, UUID providerUUID) {
         accessKey.delete();
       }
       NodeInstance.deleteByProvider(providerUUID);
+      InstanceType.deleteInstanceTypesForProvider(provider, config);
```

Comment:
So, the way we assign instance types to providers is by using the provider code, and not the provider uuid. So while for most clouds, we just allow one provider, this is not true in the case of k8s. For k8s, we can have multiple providers, and they all share instance types. So, if we always just delete the instances, some providers will end up without associated instance types. And eventually, we will be allowing multiple providers for all clouds. So in this change, first verify that there is no other provider with the same code. If there is, do not delete the instances, and if there isn't, then you can go ahead and delete the instances.

Reply:
Also, update the description of the issue as mentioned. Add sections for
Summary:
Testing:
(and then the output of the **full** sbt test command)